barretenberg/cpp/src/barretenberg/dsl/acir_format/acir_integration.test.cpp

```diff
@@ -0,0 +1,335 @@
+
+    // Function to check if a file exists
+    bool file_exists(const std::string& path)
+    {
+        std::ifstream file(path);
+        return file.good();
+    }
+
+    acir_format::AcirProgramStack get_program_stack_data_from_test_file(const std::string& test_program_name)
+    {
+        std::string base_path = "../../acir_tests/acir_tests/" + test_program_name + "/target";
+        std::string bytecode_path = base_path + "/program.json";
+        std::string witness_path = base_path + "/witness.gz";
+
+        return acir_format::get_acir_program_stack(bytecode_path, witness_path);
+    }
+
+    acir_format::AcirProgram get_program_data_from_test_file(const std::string& test_program_name)
+    {
+        auto program_stack = get_program_stack_data_from_test_file(test_program_name);
+        ASSERT(program_stack.size() == 1); // Otherwise this method will not return full stack data
+
+        return program_stack.back();
+    }
+
+    template <class Flavor> bool prove_and_verify_honk(Flavor::CircuitBuilder& builder)
+    {
+        using Prover = UltraProver_<Flavor>;
+        using Verifier = UltraVerifier_<Flavor>;
+        using VerificationKey = Flavor::VerificationKey;
+
+        Prover prover{ builder };
+        // builder.blocks.summarize();
+        // info("num gates          = ", builder.get_num_gates());
+        // info("total circuit size = ", builder.get_total_circuit_size());
+        // info("circuit size       = ", prover.instance->proving_key.circuit_size);
+        // info("log circuit size   = ", prover.instance->proving_key.log_circuit_size);
+        auto proof = prover.construct_proof();
+
+        // Verify Honk proof
+        auto verification_key = std::make_shared<VerificationKey>(prover.instance->proving_key);
+        Verifier verifier{ verification_key };
+
+        return verifier.verify_proof(proof);
+    }
+};
+
+class AcirIntegrationSingleTest : public AcirIntegrationTest, public testing::WithParamInterface<std::string> {
+  protected:
+    static void SetUpTestSuite() { srs::init_crs_factory("../srs_db/ignition"); }
+};
+
+class AcirIntegrationFoldingTest : public AcirIntegrationTest, public testing::WithParamInterface<std::string> {
+  protected:
+    static void SetUpTestSuite() { srs::init_crs_factory("../srs_db/ignition"); }
+};
+
+TEST_P(AcirIntegrationSingleTest, ProveAndVerifyProgram)
+{
+    using Flavor = GoblinUltraFlavor;
+    using Builder = Flavor::CircuitBuilder;
+
+    std::string test_name = GetParam();
+    info("Test: ", test_name);
+    acir_format::AcirProgram acir_program = get_program_data_from_test_file(test_name);
+
+    // Construct a bberg circuit from the acir representation
+    Builder builder = acir_format::create_circuit<Builder>(acir_program.constraints, 0, acir_program.witness);
+
+    // Construct and verify Honk proof
+    EXPECT_TRUE(prove_and_verify_honk<Flavor>(builder));
+}
+
+// TODO(https://github.com/AztecProtocol/barretenberg/issues/994): Run all tests
```

Comment:
Note that a handful of these don't pass--I think the cause the circuits are super tiny? Seems unimportant for now.